src/Features/CSharp/Portable/ExtractLocalFunction/CSharpExtractLocalFunctionCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod;
+using Microsoft.CodeAnalysis.ExtractMethod;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ExtractLocalFunction
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp,
+        Name = PredefinedCodeRefactoringProviderNames.ExtractLocalMethod), Shared]
+    internal class CSharpExtractLocalFunctionCodeRefactoringProvider : ExtractMethodCodeRefactoringProvider
```

Comment:
i think this was mentioned before, but please don't have a concrete provide derive from anotehr concrete provider.

Reply:
i think it would make sense to just update ExtractMethodCodeRefactoringProvider to just have the local-function-processing logic.  the base class can do something like get the ISyntaxFactsService and call a new property like ".SuppotsLocalFunctionDeclaration".

This is what several other refactorings do.  It means we have one core piece of code, but it can simply turn off/on part of the feature for C#/VB appropriately.